### PR TITLE
task-1-STREAMS development

### PR DESCRIPTION
**Pull request by Valery1508**

### CODE REVIEW

File: task-1-STREAMS/sources/Student.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class Student{
+
+    Map<String, Integer> rating;
```
Comment (IlyaBritkov) [inline task-1-STREAMS/sources/Student.java]: think the fields should be private

Comment (Valery1508) [reply task-1-STREAMS/sources/Student.java]: this task was from Ilya, so I've just copied it

Comment (IlyaBritkov) [reply task-1-STREAMS/sources/Student.java]: Maybe fix it anyway because of best practices and java convention reasons

**Comment by IlyaBritkov** [comment]
What do you think about adding a message, that is possible to use libraries like Lombok?

**Comment by IlyaBritkov** [comment]
Is the "Task" section completed in the file Tasks.md? If yes, there is the empty second point

File: task-1-STREAMS/materials/Materials.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Materials
+
+## Links:
```
Comment (IlyaBritkov) [inline task-1-STREAMS/materials/Materials.md:3]: Maybe add some comments about article content. For example:
Java 8 Stream API: [link]

File: task-1-STREAMS/Tasks.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# LeverX Stream Task
+
+## Important information:
+1) You can find some useful materials in [Materials.md](materials/Materials.md);
+2) Pay attention to the file [Self-check.md](Self-check.md);
+3) You are able to use Lombok.
```
Comment (IlyaBritkov) [inline task-1-STREAMS/Tasks.md]: Add to 3) point: 
"3) You are able to use Lombok.and object-mapping libraries"

Also maybe add one more point:
"
4) Don't write all code in one class. Use service layer for business logic and feel free to divide logic by methods.
"

File: task-1-STREAMS/Tasks.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    ```
+5) Group models from 2 task by any 4 parameters.
+6) Considering strings such as: a = "aaabbxxxxxtx" b = "p1p1p1p". Implement func(string) that returns the most common character, and the number of times it appears in the string passed as single argument.
+    ``` 
+    print func(a)
+    ["x", 6]
+    print func(b)
+    ["p", 4]
+    ```
+7) Find best matching person. For example, if someone is looking for English with more than 50% proficiency then we need to search for Person whose proficiency is more than 50%. In the above example it is Gnanesh who has 100% proficiency in English. If none of the Person meets Given threshold (Case for Kannada in above example) then find Technician who is best among rest ie Suresh in the above case (40%). System should be bookkeeping i.e. if we book a slot then it must not be offered again. Use [this](sources/Person.txt) file.
```
Comment (IlyaBritkov) [inline task-1-STREAMS/Tasks.md]: "In the above example"? Where? I suppose in the file, but such description can confuse

File: task-1-STREAMS/sources/Task.java [hunk 5]
```diff
@@ -0,0 +1,9 @@
+public class Task {
+    private final String id;
+    private final String title;
+    private final TaskType type;
+    private final LocalDate createdOn;
+    private boolean done = false;
```
Comment (IlyaBritkov) [inline task-1-STREAMS/sources/Task.java]: Field "done" is false by default

**Review by IlyaBritkov — CHANGES_REQUESTED** [review]

File: task-1-STREAMS/Tasks.md [hunk 6]
```diff
@@ -0,0 +1,29 @@
+# LeverX Stream Task
+
+## Important information:
+1) You can find some useful materials in [Materials.md](materials/Materials.md);
+2) Pay attention to the file [Self-check.md](Self-check.md);
+3) You are able to use Lombok.
+
+## Task description:
```
Comment (IlyaBritkov) [inline task-1-STREAMS/Tasks.md:9]: Let's rename this title to "Requirements"

File: task-1-STREAMS/Tasks.md [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+You have to implement **tasks given below** using **Java 8+ Stream API**.
+
+## Tasks:
+1) There is an ArrayList with objects from class [Student](sources/Student.java) . Calculate an average score for a specific subject. Fill in the collection yourself.
```
Comment (Shotim) [inline task-1-STREAMS/Tasks.md]: just "List"

File: task-1-STREAMS/materials/Materials.md (hunk 2 above)
Comment (Shotim) [reply task-1-STREAMS/materials/Materials.md:3]: @IlyaBritkov If think it's redundant

**Review by Shotim — CHANGES_REQUESTED** [review]

File: task-1-STREAMS/sources/Task.java (hunk 5 above)
Comment (Valery1508) [reply task-1-STREAMS/sources/Task.java]: @IlyaBritkov And?

File: task-1-STREAMS/Tasks.md (hunk 6 above)
Comment (Valery1508) [reply task-1-STREAMS/Tasks.md:9]: @Shotim @IlyaBritkov but it is not requirements...

File: task-1-STREAMS/sources/Task.java (hunk 5 above)
Comment (IlyaBritkov) [reply task-1-STREAMS/sources/Task.java]: get rid of initialization

File: task-1-STREAMS/Tasks.md (hunk 6 above)
Comment (IlyaBritkov) [reply task-1-STREAMS/Tasks.md:9]: Let's just call it like this. Additionally, there could be exactly requirements in the future

**Review by Shotim — APPROVED** [review]

**Pull request merged**